src/components/Player/Player.tsx

```diff
@@ -0,0 +1,157 @@
+import React, { useCallback, useEffect, useRef, useState } from 'react';
+
+import styles from './Player.module.scss';
+
+import { addScript } from '#src/utils/dom';
+import { deepCopy } from '#src/utils/collection';
+import type { JWPlayer } from '#types/jwplayer';
+import type { PlaylistItem } from '#types/playlist';
+import useEventCallback from '#src/hooks/useEventCallback';
+import useOttAnalytics from '#src/hooks/useOttAnalytics';
+import { logDev } from '#src/utils/common';
+
+type Props = {
+  playerId?: string | null;
+  feedId?: string;
+  item: PlaylistItem;
+  onReady?: (player?: JWPlayer) => void;
+  onPlay?: () => void;
+  onPause?: () => void;
+  onComplete?: () => void;
+  onUserActive?: () => void;
+  onUserInActive?: () => void;
+  onBeforePlay?: () => void;
+  onPlaylistItemCallback?: (item: PlaylistItem) => Promise<undefined | PlaylistItem>;
+  startTime?: number;
+};
+
+const Player: React.FC<Props> = ({
+  playerId,
+  item,
+  onReady,
+  onPlay,
+  onPause,
+  onComplete,
+  onUserActive,
+  onUserInActive,
+  onBeforePlay,
+  onPlaylistItemCallback,
+  feedId,
+  startTime = 0,
+}: Props) => {
+  const playerElementRef = useRef<HTMLDivElement>(null);
+  const playerRef = useRef<JWPlayer>();
```

Comment:
```suggestion
  const playerRef = useRef<JWPlayer>();
  const player = playerRef.current; // Use this everywhere instead of playerRef.current (except for the assignment call) and code inspections work better to pick up missing dependencies for callbacks, etc.
```